### ISSUE TITLE
Ensure "Clicked on Create Account" events register properly on DEV

### ISSUE THREAD

**Issue by pkfrank** [forem/forem#18503]
**Describe the bug**

The recently merged PR here (https://github.com/forem/forem/pull/18459) added tracking to the registration flow.  However, on DEV, it doesn't appear to be passing through all of the intended properties into the ahoy_event.  This may be due to a DEV-specific configuration that isn't present in local development.

**To Reproduce**

1. Take an action on DEV that should register an event with a trigger and referring source
2. View recent Ahoy events (requires access to DEV Production database)
3. See that some fields are missing (such as "trigger" and "referring_source")

**Expected behavior**

All properties described in this PR (https://github.com/forem/forem/pull/18459) register in the ahoy_events on all production environments, including on DEV.

**Additional context**

Core Team with access to Metabase can view the relevant recent events [here](https://forem.metabaseapp.com/question/186-ahoy-events-for-clicked-on-create-account).

**Comment by benhalpern** [comment]
My initial guess is that this is related to our Fastly go-between. If the visit token is generated by or related to the IP address Ahoy is expecting, we may obfuscate the typical answer because all our traffic runs through the edge.

That's why we have code like this:

```ruby
  def anonymous_user
    User.new(ip_address: request.env["HTTP_FASTLY_CLIENT_IP"] || request.env["HTTP_X_FORWARDED_FOR"])
  end
```

This isn't deeply investigated, but my initial hunch. @maestromac might have some more knowledge/ideas here.

**Comment by Ridhwana** [comment]
@pkfrank I spent some time looking at the stats this morning and I'm not sure that I'm seeing the issue that you've outlined above. 

I've created a [new metabase query](https://forem.metabaseapp.com/question/188-ahoy-tracking-clicked-on-create-account-events). This query shows all the "Clicked on Create Account" events. 

To make sure that we start of with the same assumptions I'm just going to summarize the expected behaviour from [this post ](https://forem.team/ridhwana/tracking-of-registration-workflow-clicked-on-create-account-1cee). There are two types of registration events:

- The first type being "Create Account" buttons that are directly shown on the pages to explicitly nudge the user to create an account. In this case we expose two properties, the `source` and the `page`. The page will contain the URL of the page that the user clicked the button from, this will help us to distinguish whether the button was on a home page or perhaps
an article page. The source helps us to find which "Create Account" button was clicked, was it in the sidebar, or in the top
nav etc.

- The second type is when a user wants to be able to interact with content, they are prompted indirectly to create an account through a modal. In this case, we need a few more properties to track where the trigger originated from. The `source` property will always be a modal for these. However, in order to determine what triggered the modal, we have added two other properties, a `referring_source` which will outline where the action was triggered from, for example the reaction_toolbar and a `trigger` property that will narrow down the exact trigger, for example a like or readinglist. The page property will contain the URL of the page from which the user clicked the button. 

As I look through the ["Clicked on Create Account"](https://forem.metabaseapp.com/question/188-ahoy-tracking-clicked-on-create-account-events) events, I see all the properties being listed accurately. 

A direct "Create Account" click logs the `page` and `source`, but when a click comes from a source `modal` (i.e. some interaction) I'm seeing the `referring_source` and the `trigger` being listed alongside as well. 
I created a query on metabase that shows the "Clicked on Create Account" events via the modal which you can find [here](https://forem.metabaseapp.com/question/189-ahoy-tracking-clicked-on-create-account-events-only-within-modals). 

Is the issue that you're seeing intermittent, or across all data? Had you perhaps thought a `referring_source` or `trigger` was to appear on non-modal events and you were referring to those rows?

**Comment by mirie** [comment]
Added blocked label as waiting for clarification

**Comment by pkfrank** [comment]
Hi @Ridhwana , thanks for this reply, and for the further information + research.

**You are right — this does appear to tracking properly.**  I'm sorry for sending you down a distracting hunt for a non-issue.

If you're curious, my confusion was a function of:

1) Precisely what you guessed here

> Had you perhaps thought a referring_source or trigger was to appear on non-modal events and you were referring to those rows?

2) The way I set up my original Metabase dashboard required me to click into each row to view the full list of properties (whereas your native SQL version displays all of the properties on the page), so I couldn't `ctrl+f` for the `"trigger"` keyword

3) My assumption on the relative frequency of events from "direct click" vs. "trigger based click" was way off-base, in that "trigger" clicks make up only a small fraction (I thought it would have been much more even, which caused me to give up on my sampling too early).

So, TLDR, I think this is ready to be closed, and apologies for the confusion and back-tracking here.

On a separate note -- but something that is visible from your Metabase query -- it appears that a certain `visit_id` represents a huge % of the total events.  Will you give some consideration into whether that's an issue worth pursuing while this introduction is still fresh in mind?

**Comment by pkfrank** [comment]
Per most recent comments, I'll go ahead and close this issue.
